src/lsm/groove.zig

```diff
@@ -946,6 +947,13 @@ pub fn GrooveType(
                     }
                 }
             }
+
+            // Putting the objects_cache upsert after the index tree updates is critical:
+            // We diff the old and new objects, but the old object will be a pointer into the
+            // objects_cache. If we upsert first, old.* == new.* and no secondary indexes will
+            // be updated!
+            groove.objects_cache.upsert(new);
+            groove.objects.put(new);
```

Comment:
For an extra tripple check, can we assert here that now old & new _contents_ is the same?

Reply:
Ok - so that won't actually hold; if the old value comes from the stash of the cache_map, and we insert a new value, it'll go into the cache - so now `old.* != new.*`. 

The assertion at the start, `assert(old_from_cache == old);` will still hold because if it comes from the stash from the caller, it should still come from the stash inside.